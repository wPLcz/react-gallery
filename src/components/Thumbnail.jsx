import React, { useEffect, useRef } from 'react';
import styled, { keyframes } from 'styled-components';
import PropTypes from 'prop-types';
import { v4 as uuidv4 } from 'uuid';

const Thumbnail = (props) => {

  return (
    <Figure
      onClick={() => props.handleClickForDetails(props.id)}
    >
      <Img src={props.src} alt={props.alt}/>
      <Figcaption>
        <ListContainer>
          {props.location && <li>Country: {props.location}</li>}
          {props.likes && <li>Likes number: {props.likes}</li>}
          {props.downloads && <li>Downloads: {props.downloads}</li>}
        </ListContainer>
      </Figcaption>
    </Figure>
  )
};

const Img = styled.img`
    transition: transform 2s ease-out;
    width: 120%;
`;

const ListContainer = styled.ul`
    position: absolute;
    left:0;
    bottom: 0;
    list-style-type: none;
    font-weight: 300;
    font-size: 2rem;
    font-family: Helvetica;
    color: white;
`;

const Figcaption = styled.figcaption`
    margin: 0 auto;
    bottom: 0;
    left: 0;
    width: 100%;
    height: 0%;
    opacity: 0;
    display: block;
    position: absolute;
    transition: height 350ms ease-in-out, opacity 250ms ease-in-out 150ms;
    background-color: rgba(68, 72, 79, 0.4);
    align-self: center;
`;

const Figure = styled.figure`
    max-height: 300px;
    display: grid;
    position: relative;
    border-radius: 6px;
    overflow: hidden;
    cursor: pointer;
    justify-content: center;
    align-content: center;
    background-color: black;
    box-shadow: 0px 0px 5px 1px rgba(87,86,87,1);
    
    
    &:hover ${Img} {
      transform: scale(1.25);
    }
   
    &:hover ${Figcaption} {
      height: 100%;
      opacity: 1;
    }
    @media (min-width: 960px) {
      max-height: 350px;
    }
  
    @media (min-width: 1300px) {
      max-height: 400px;
    }
`;

Thumbnail.propTypes = {
  id: PropTypes.string,
  delayFactor: PropTypes.number,
  src: PropTypes.string,
  alt: PropTypes.string,
};

Thumbnail.defaultProps = {
  id: uuidv4(),
  delayFactor: 1,
  src: '',
  alt: '',
};

export default Thumbnail;
