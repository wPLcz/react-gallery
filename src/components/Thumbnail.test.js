import React from "react";
import ReactDOM from "react-dom";
import renderer from 'react-test-renderer';
import Thumbnail from "./Thumbnail";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Thumbnail/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly when there are no items', () => {
  const tree = renderer.create(<Thumbnail/>).toJSON();
  expect(tree).toMatchSnapshot();
});
