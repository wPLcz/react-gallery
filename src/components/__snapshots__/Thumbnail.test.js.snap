// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly  1`] = `
<figure
  className="sc-hKFyIo imHzss"
  onClick={[Function]}
>
  <img
    alt=""
    className="sc-bdnylx kVcZjp"
    src=""
  />
  <figcaption
    className="sc-dlnjPT dXIDcK"
  >
    <ul
      className="sc-gtssRu olpFb"
    />
  </figcaption>
</figure>
`;

exports[`renders correctly when there are no items 1`] = `
<figure
  className="sc-hKFyIo imHzss"
  onClick={[Function]}
>
  <img
    alt=""
    className="sc-bdnylx kVcZjp"
    src=""
  />
  <figcaption
    className="sc-dlnjPT dXIDcK"
  >
    <ul
      className="sc-gtssRu olpFb"
    />
  </figcaption>
</figure>
`;
